navigation/navigation-api/src/main/java/com/duckduckgo/navigation/api/GlobalActivityStarter.kt

```diff
@@ -0,0 +1,129 @@
+import java.io.Serializable
+import kotlin.jvm.Throws
+
+/**
+ * This is the Activity Starter.
+ * Use this type to start Activities or get their start intent.
+ *
+ * Activities to launch are identified by their input [ActivityParams].
+ *
+ * ```kotlin
+ * data class ExampleActivityParams(...) : ActivityParams
+ *
+ * globalActivityStarter.start(context, ExampleActivityParams(...))
+ * ```
+ */
+interface GlobalActivityStarter {
+
+    /**
+     * Starts the activity given its [params].
+     *
+     * The activity can later retrieve the [Serializable] [params] using the extension function [Bundle.getActivityParams]
+     *
+     * @param params The activity parameters. They also identify the activity
+     * @param context the context used to start the activity
+     * @param options additional options for how the activity should be started, eg. scene transition animations
+     *
+     * @throws IllegalArgumentException when the Activity can't be found
+     */
+    @Throws(IllegalArgumentException::class)
+    fun start(context: Context, params: ActivityParams, options: Bundle? = null)
+
+    /**
+     * Returns  the [Intent] that can be used to start the [Activity], given the [ActivityParams].
+     * This method will generally be used to start the activity for result.
+     *
+     * @return the [Intent] that can be used to start the [Activity]
+     */
+    fun startIntent(context: Context, params: ActivityParams): Intent?
+
+    /**
+     * This is a marker class
+     * Mark all data classes related to activity arguments with this type
+     */
+    interface ActivityParams : Serializable
+
+    /**
+     * Implement this mapper that will return [Activity] class for the given parameters.
+     * Once implemented it, you need to contribute it as a multibinding using [ContributesMultibinding] into the [AppScope].
+     * ```kotlin
+     * @ContributesMultibinding(AppScope::class)
+     * class ExampleParamToActivityMapper @Inject constructor(...) : ParamToActivityMapper {
+     *   fun fun map(params: ActivityParams): Class<out AppCompatActivity>? {
+     *     return if (params is ExampleActivityParams) {
+     *       ExampleActivity::class.java
+     *     }
+     *     else {
+     *       null
+     *     }
+     *   }
+     * }
+     * data class ExampleActivityParams(...) : ActivityParams
+     *
+     * class ExampleActivity() : DuckDuckGoActivity() {...}
+     * ```
+     *
+     * Alternatively you can also use the [ContributeToActivityStarter] annotation to autogenerate the parap to activity mapper above.
+     * ```kotlin
+     * @ContributeToActivityStarter(ExampleActivityParams::class)
+     * class ExampleActivity() : DuckDuckGoActivity() {...}
+     * ```
+     */
+    interface ParamToActivityMapper {
+        /**
+         * @return the [Activity] class if it matches the desired parameters, otherwise return `null`
+         */
+        fun map(activityParams: ActivityParams): Class<out AppCompatActivity>?
```

Comment:
should this be `DuckDuckGoActivity` instead? We would avoid adding app-compat as dependency to the module.